components/brave_wallet/browser/asset_ratio_service.cc

```diff
@@ -154,6 +170,53 @@ GURL AssetRatioService::GetPriceHistoryURL(
   return GURL(spec);
 }
 
+void AssetRatioService::GetBuyUrlV1(mojom::OnRampProvider provider,
+                                    const std::string& chain_id,
+                                    const std::string& address,
+                                    const std::string& symbol,
+                                    const std::string& amount,
+                                    const std::string& currency_code,
+                                    GetBuyUrlV1Callback callback) {
+  std::string url;
+  if (provider == mojom::OnRampProvider::kWyre) {
+    url =
+        base::StringPrintf(kWyreBuyUrl, address.c_str(), currency_code.c_str(),
```

Comment:
what about url encoding for these values? This seems fragile to me, I think it would be better to use something like `url::Component` here

Reply:
actually `AppendQueryParameter`

Reply:
Fixed in this pull request and opened https://github.com/brave/brave-browser/issues/24401 to investigate other cases.